src/udiskslinuxdriveata.c

```diff
@@ -470,6 +470,36 @@ static gboolean get_pm_state (UDisksLinuxDevice *device, GError **error, guchar
   return rc;
 }
 
+/**
+ * udisks_linux_drive_ata_is_active:
+ * @device: The ata device to get pm state from
+ * @error: Return location for error or %NULL
+ *
+ * Get the current power mode status from a drive.
+ *
+ * The format of @state is the result obtained from sending the
+ * ATA command CHECK POWER MODE to the drive.
+ * Known values include:
+ *     0x00 Standby
+ *     0x80 Idle
+ *     0xff Active/Idle
+ * Typically user interfaces will report "Drive is spun down" if @state is
+ * 0x00 and "Drive is spun up" otherwise.
+ *
+ * Returns: %FALSE if the drive is spun down, %TRUE otherwise.
+ */
+gboolean udisks_linux_drive_ata_is_active (UDisksLinuxDevice *device, GError **error)
+{
+  gboolean awake = FALSE;
```

Comment:
Here again with the _FALSE_ default value you wouldn't distinguish between the ATA command failure and device sleeping. Since this is a specific call, please default to _TRUE_ instead. We only want to skip further calls when _state_ is properly populated and indicates the drive is sleeping.

I think you should remove the `error` argument as well since it's not tested further in the code and the function return value semantics doesn't correspond to the usual GLib conventions anyway.

What I originally had in mind was that this `udisks_linux_drive_ata_is_active()` function (or a macro ideally) wouldn't do any I/O and just serve as a convenient function analyzing the _state_ value. In that case you would still have to call `get_pm_state()` as usual, thus there wouldn't be any issue to distinguish ATA command failure and the value.

Reply:
Thanks for the feedback. I like that educational approach.
Please take a look again.